airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/sonar_cloud.yaml

```diff
@@ -0,0 +1,102 @@
+version: "0.1.0"
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: []
+  components_selector:
+    extractor:
+      field_pointer: ["components"]
+  issues_selector:
+    extractor:
+      field_pointer: ["issues"]
+  metrics_selector:
+    extractor:
+      field_pointer: ["metrics"]
+  requester:
+    url_base: "https://sonarcloud.io/api"
+    http_method: "GET"
+    request_options_provider:
+      request_parameters:
+        organization: "{{ config['organization'] }}"
+        createdAfter: "{{ config['start_date'] }}"
+        createdBefore: "{{ config['end_date'] }}"
+    authenticator:
+      type: BearerAuthenticator
+      api_token: "{{ config['user_token'] }}"
+  increment_paginator:
+    type: "DefaultPaginator"
+    url_base: "*ref(definitions.requester.url_base)"
+    page_size_option:
+      inject_into: "request_parameter"
+      field_name: "ps"
+    pagination_strategy:
+      type: "PageIncrement"
+      page_size: 100
+    page_token_option:
+      inject_into: "request_parameter"
+      field_name: "p"
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
+    requester:
+      $ref: "*ref(definitions.requester)"
+  components_retriever:
+    record_selector:
+      $ref: "*ref(definitions.components_selector)"
+    paginator:
+      $ref: "*ref(definitions.increment_paginator)"
+    requester:
+      $ref: "*ref(definitions.requester)"
+  issues_retriever:
+    record_selector:
+      $ref: "*ref(definitions.issues_selector)"
+    paginator:
+      $ref: "*ref(definitions.increment_paginator)"
+    requester:
+      $ref: "*ref(definitions.requester)"
+  metrics_retriever:
+    record_selector:
+      $ref: "*ref(definitions.metrics_selector)"
+    paginator:
+      $ref: "*ref(definitions.increment_paginator)"
+    requester:
+      $ref: "*ref(definitions.requester)"
```

Comment:
You can simplify this after having only one extractor.

Reply:
Done!